plugins/woocommerce/src/Admin/DataSourcePoller.php

```diff
@@ -133,7 +133,10 @@ public function read_specs_from_data_sources() {
 			$this->merge_specs( $specs_from_data_source, $specs, $url );
 		}
 
-		$specs_group            = get_transient( $this->args['transient_name'] ) ?? array();
+		$specs_group = get_transient( $this->args['transient_name'] );
+		if ( false === $specs_group ) {
```

Comment:
Hit the same problem, happy to find an existing PR that addresses it! 

This fix does indeed solve the problem for me locally, although I wonder if the following might be a better (by better I guess I mean, slightly more defensive) strategy:

```suggestion
		if ( ! is_array( $specs_group ) ) {
```

My thinking here is that the problem I hit, and that probably @markjaquith hit, was that the transient was not set and so we got back `(bool) false` where we expected an array ... likely that will be the most common form of the problem, so we can treat my suggestion here as purely optional.

However, the point is that transients can be set to anything—and if it was set to a string, for instance, we'd get a fatal uncaught type-error and not just a deprecation notice. So, really what we want to do here is ensure we have an array (ie, check that the transient is set, and that it is set to the correct type).